RunStuff Template.m

```diff
@@ -256,37 +257,74 @@
 \[Phi]1=Sum[(munits ndensities[datasr,"xflavor"-> False][[1,i,i]]-munits ndensities[datasr,"xflavor"-> False][[2,i,i]])\[Theta][[i]],{i,1,Length[\[Theta]]}];
-Ve=(munits/mp) *datasr["Yes"] *datasr["matters"];
+
 \[CapitalOmega]p=Table[N[evs[[i]]-Ve-\[Phi]0],{i,1,Length[evs]}];
+
 kp=k-\[Phi]1;
 \[Omega]=\[Omega]EMev[Ener];
-Print[Length[evs]];
-Print[Length[\[CapitalOmega]p]];
-Print[Length[\[Theta]]];
-Print[Ve];
-Print[\[CapitalOmega]p];
-Print[evs];
-Print[N[\[CapitalOmega]p[[1]]-kp \[Theta][[1]]]];
-tab=Table[{\[CapitalOmega]p[[i]],\[Theta][[j]],N[\[CapitalOmega]p[[i]]-kp \[Theta][[j]]]},{i,1,Length[\[CapitalOmega]p]},{j,1,Length[\[Theta]]}];
+
 checks=Table[{evs[[i]],dispersionCheck[datasr,evs[[i]],k]},{i,1,Length[evs]}];
-Print[Table[N[(munits(ndensities[datasr,"xflavor"-> False][[1]][[i,i]]- ndensities[datasr,"xflavor"-> False][[2]][[i,i]])),40],{i,1,Length[\[Theta]]}]];
-values={\[Phi]0,\[Phi]1,Ve,kp,\[Omega],data["radius"][[250]]};
-Print[\[Phi]0];Print[\[Phi]1];Print[data["radius"][[250]]];
-Return[{checks//MatrixForm,tab//MatrixForm}];
 
+bottoms=Table[(\[CapitalOmega]p[[i]]-(kp \[Theta][[j]])),{i,1,Length[evs]},{j,1,Length[\[Theta]]}];
+
+values={"\[Phi]0",\[Phi]0,"\[Phi]1",\[Phi]1,"Ve",Ve,"k",k,"kp",kp,"\[Omega]",\[Omega],"r",data["radius"][[r]]};
+
+Return[{esys,values,checks,bottoms}];
+
+
+];
+
+
+evecviz[eigensys_,ni_,labels_]:=Module[{n,nulen,nublen,data,c\[Theta],nuarrow,nubarrow,draw,plotout,gridlines,gridlabels},
+n=Length[eigensys[[2]]];
+data=ImportData["G:\\My Drive\\Physics\\Neutrino Oscillation Research\\Fast Conversions\\lotsadata.tar\\lotsadata\\lotsadata\\112Msun_100ms_DO.h5"];
+c\[Theta]=data["mids"]; (*cos{theta}*)
+nulen[i_]:=Normalize[eigensys[[2,i,1;;10]]]//Abs//Chop ;(*neutrino eigenvecot rlength for eigenvector i.  Normalized, and abs as some small parts go negative.  Chop rounds very small parts down to 0*)
+nublen[i_]:=Normalize[eigensys[[2,i,11;;20]]]//Abs//Chop ; (*anti-neutrino eigenvecot rlength for eigenvector i.  Normalized, and abs as some small parts go negative.*)
+nuarrow[i_]:=Table[Arrow[{{0,0},{nulen[i][[j]] c\[Theta][[j]],nulen[i][[j]] Sin[ArcCos[c\[Theta][[j]]]] }}],{j,1,Length[c\[Theta]]}] ;(*draws an arrow for the vector*)
+nubarrow[i_]:=Table[Arrow[{{0,0},{nublen[i][[j]] c\[Theta][[j]],nublen[i][[j]] Sin[ArcCos[c\[Theta][[j]]]]}}],{j,1,Length[c\[Theta]]}];
+gridlines=Table[Line[{{0,0},{ c\[Theta][[j]], Sin[ArcCos[c\[Theta][[j]]]] }}],{j,1,Length[c\[Theta]]}]; (*Draws line segments for the angular bin mids*)
+gridlabels=Table[Text[labels[[j]],{c\[Theta][[j]], Sin[ArcCos[c\[Theta][[j]]]]}],{j,1,Length[c\[Theta]]}]; 
+draw[i_]:=Graphics[{Black,Dotted,gridlines,Blue,Arrowheads[0.02],nuarrow[i],Red,Arrowheads[0.02],Dashed,nubarrow[i],Purple,gridlabels},Axes->True,PlotRange-> {{-1,1},{0,1}},PlotRangePadding-> 0.15,PlotLabel->eigensys[[1,i]],ImageSize-> Scaled[0.5]];
+Return[draw[ni]]
 ];
 
 
+testsys=listDispersion[280];

```

Comment:
This looks like scratch work. Should we delete it from the repository?

Reply:
Not scratch work, but definitely some "debug" work.  I'll move this to a new working notebook, as it doesn't need to be in the checkfile.